src/aws/osml/model_runner/database/ddb_helper.py

```diff
@@ -104,6 +105,40 @@ def put_ddb_item(self, ddb_item: DDBItem, condition_expression: str = None) -> D
 
         return response
 
+    def batch_write_items(self, ddb_items: List[DDBItem], max_retries: int = 40, max_delay: float = 8):
+        """
+        Write multiple DynamoDB items in a batch with retry logic for unprocessed items.
+        :param ddb_items: List[DDBItem] = List of items that we want to write.
+        :param max_retries: int = Maximum number of retries for unprocessed items.
+        :param max_delay: Maximum delay in seconds between retries.
+        :return: None
+        """
+
+        def _batch_write(items: Dict[str, Any], retries=0):
+            response = self.client.batch_write_item(RequestItems=items)
```

Comment:
It doesn't look like there is any error handling if the boto3 client throws an exception. Is that intentional? Is the idea to fail the entire batch / tile / region if one of these batch_write_item calls fails or should that be caught up and included in the retry loop for this function?

Reply:
Also consider using boto3's batch_writer. It seems to implement a bunch of this logic already.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb/table/batch_writer.html

Reply:
I did play with implementing that batch_writer() piece early on and it was giving me problems when it came to the auto retrying of unprocessed items - but I will see if I can try it again.

Reply:
I am going to hold off on handling these errors elegantly because no matter what if the region fails the entire region would have to be retried anyway.... so might as well fail it early.